src/Microsoft.TestPlatform.CrossPlatEngine/TestSession/TestSessionPool.cs

```diff
@@ -24,32 +25,37 @@ public class TestSessionPool
         /// <summary>
         /// Initializes a new instance of the <see cref="TestSessionPool"/> class.
         /// </summary>
-        private TestSessionPool()
+        internal TestSessionPool()
         {
             this.sessionPool = new Dictionary<TestSessionInfo, ProxyTestSessionManager>();
```

Comment:
Yep, exactly this, too much wiring needed to get things going. Also, I don't foresee a case in which we'd want to have multiple `TestSessionPool` instances because we already can manage all the sessions corresponding to an instance of `vstest.console` using the same `TestSessionPool` instance.